js/data/bucket/symbol_bucket.js

```diff
@@ -184,6 +192,12 @@ class SymbolBucket {
             // It's better to place labels on one long line than on many short segments.
             this.features = mergeLines(this.features);
         }
+
+        // set up this mapping to allow for constant-time lookups in place()
+        this.featuresByIndex = {};
```

Comment:
@jfirebaugh the symbolInstance is being stored (at `populate()` time) in a StructArray (`this.symbolInstancesArray`), so we can't attach a reference.

Granted, (a) we're not yet using `symbolInstancesArray` on a different thread, and (b) once we do (for cross-source placement), we'll have to find a different way to handle the paint attributes, since presumably whichever thread gets the array won't have the property data from the source.